src/QAT/qat/purr/integrations/tket.py

```diff
@@ -366,12 +366,16 @@ def run_tket_optimizations(qasm_string, opts, hardware: QuantumHardwareModel) ->
         )
         return qasm_string
 
-    couplings = hardware.qubit_direction_couplings
+    couplings = hardware.qubit_direction_couplings.copy()
```

Comment:
why copy?

Reply:
Swapped to `deepcopy`. It's because we do not want to actually change the couplings on the `hardware model`.